core/optaplanner-constraint-streams/src/main/java/org/optaplanner/constraint/streams/bavet/common/index/Indexer.java

```diff
@@ -64,9 +67,31 @@
     Value_ remove(Object[] indexProperties, Tuple_ tuple);
 
     /**
+     * @param indexProperties never null
+     * @param visitor never null
+     */
+    void visit(Object[] indexProperties, BiConsumer<Tuple_, Value_> visitor);
+
+    /**
+     * @param indexProperties never null
+     * @return number of values across all tuples
+     */
+    default int countValues(Object[] indexProperties) {
+        AtomicInteger counter = new AtomicInteger();
+        visit(indexProperties, (tuple, value) -> counter.incrementAndGet());
+        return counter.get();
+    }
+
+    /**
+     * For iteration, prefer the use of {@link #visit(Object[], BiConsumer)} as it does not create any intermediate {@link Map}.
+     *
      * @param indexProperties never null
      * @return never null
      */
-    Map<Tuple_, Value_> get(Object[] indexProperties);
+    default Map<Tuple_, Value_> get(Object[] indexProperties) {
+        Map<Tuple_, Value_> result = new LinkedHashMap<>();
+        visit(indexProperties, result::put);
+        return result;
+    }
```

Comment:
I keep this method as the tests are designed to use it; but there really is no other reason for its existence.
It is a killer of performance when used.

Reply:
Now even more. Let's remove this method - because we don't want any code to use its. As tests need it, we can move it into the tests?